internal/controller/controllers/bmo_utils.go

```diff
@@ -91,6 +106,37 @@ func (r *bmoUtils) GetIronicIPs() ([]string, []string, error) {
 	return ironicIPs, inspectorIPs, nil
 }
 
+func (r *bmoUtils) getICCConfig(ctx context.Context) (*ICCConfig, error) {
+	const configKeyNotFoundError string = "Failed to get '%s' key from secret %s in namespace %s"
+
+	secret := &corev1.Secret{}
+	namespacedName := types.NamespacedName{Name: iccSecretName, Namespace: iccNamespace}
+	if err := r.c.Get(ctx, namespacedName, secret); err != nil {
+		return nil, fmt.Errorf("Failed to get secret %s in namespace %s: %w", iccSecretName, iccNamespace, err)
+	}
+
+	ironicBaseURL, ok := secret.Data[ironicBaseURLKey]
+	if !ok {
+		return nil, fmt.Errorf(configKeyNotFoundError, ironicBaseURLKey, secret.Name, secret.Namespace)
+	}
+
+	ironicInspectorBaseURL, ok := secret.Data[ironicInspectorBaseURLKey]
+	if !ok {
+		return nil, fmt.Errorf(configKeyNotFoundError, ironicInspectorBaseURLKey, secret.Name, secret.Namespace)
```

Comment:
Late to this party, sorry. This is not quite correct: the inspector URL can and actually will be missing.

Reply:
In that case, how we get the inspector URL? From the preprovisioning info?

Reply:
Ok, if understand well, we can leave the inspector URL empty, and it will be auto-magically filled using the ICC lib https://github.com/openshift/image-customization-controller/commit/26ce5d6c64ee4ff120a8f7f56eb44397ca7661f1

Reply:
This is correct.

Reply:
:confused: but we don't use any of the icc stuff, right?

Reply:
we use it here: https://github.com/openshift/assisted-service/blob/8703b59882a2f9d5730ece7c2c47cb025ab42a8f/internal/ignition/ironic.go#L22